commands/connect/diagnose.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+const api = require('../../lib/connect/api.js')
+const regions = require('../../lib/connect/regions.js')
+const cli = require('heroku-cli-util')
+const co = require('co')
+
+function displayResults (results) {
+  results.errors.forEach(displayResult('red'))
+  results.warnings.forEach(displayResult('yellow'))
+  results.passes.forEach(displayResult('green', false))
+}
+
+function displayResult (color, displayMessages) {
+  // Default to displaying messages, unless overridden
+  if (displayMessages === undefined) {
+    displayMessages = true
+  }
+  return function (result) {
+    cli.log(cli.color[color](`${color.toUpperCase()}: ${result.display_name}`))
+    if (displayMessages) {
+      cli.log(result.message)
+      if (result.doc_url) {
+        cli.log(result.doc_url)
+      }
+    }
+  }
+}
+
+module.exports = {
+  topic: 'connect',
+  command: 'diagnose',
+  description: 'Display diagnostic information about a connection',
+  help: 'Checks a connection for common configuration errors. ',
+  flags: [
+    {name: 'resource', description: 'specific connection resource name', hasValue: true},
+    regions.flag
+  ],
+  needsApp: true,
+  needsAuth: true,
+  run: cli.command(co.wrap(function * (context, heroku) {
+    context.region = regions.determineRegion(context)
+    let connection = yield api.withConnection(context, heroku)
+    let results = yield cli.action('Diagnosing connection', co(function * () {
+      let url = '/api/v3/connections/' + connection.id + '/validations'
+      return yield api.request(context, 'GET', url)
+    }))
+
+    cli.log() // Blank line to separate each section
+    cli.styledHeader(`Connection: ${connection.name}`)
+    displayResults(results.json)
+
+    for (let objectName in results.json.mappings) {
+      cli.log() // Blank line to separate each section
+      cli.styledHeader(`Mapping: ${objectName}`)
```

Comment:
Hrm. We don't really have any good options for that within `styledHeader`, so we have a couple options.

One is to roll our own heading, and just not use `styledHeader` at all. That would give us full flexibility, but we'd start to drift away from other CLI plugins by doing so.

The other option is to remove "Mapping:" altogether and just display the name of each mapping. I was worried there'd be confusion between the connection and the mappings if we don't display that identifier, but maybe it's okay?

<img width="941" alt="screen shot 2017-06-27 at 6 04 48 pm" src="https://user-images.githubusercontent.com/2522/27612128-24c0eed4-5b63-11e7-8f66-cc11bfbe0f6f.png">